### PR TITLE
Split installer policy into 2 policies (in separate files)

### PR DESCRIPTION
**Pull request by oriAdler**
### What type of PR is this?
_feature_

### What this PR does / why we need it?
Provide minimal permission in the user account.

### Which Jira/Github issue(s) this PR fixes?

_Fixes #_ https://issues.redhat.com/browse/SDA-6771

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Included documentation changes with PR
- [ ] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

Follows up: https://github.com/openshift/managed-cluster-config/pull/1300

### CODE REVIEW

**Comment by oriAdler** [comment]
@james I was able to create BYO VPC cluster 4.8.z, 4.9.z, 4.10.z with `resources/sts/4.11/sts_installer_core_permission_policy.json`.

**Comment by oriAdler** [comment]
@james I was able to create **non BYO VPC** cluster 4.9.z, 4.10.z, 4.11.z with `resources/sts/4.11/sts_installer_core_permission_policy.json` and `resources/sts/4.11/sts_installer_vpc_permission_policy.json`.

The last missing piece is the **private link** cluster, can you please confirm the permissions we added to `resources/sts/4.11/sts_installer_privatelink_permission_policy.json`?

File: resources/sts/4.11/sts_installer_privatelink_permission_policy.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+		    "ec2:ModifyVpcEndpointServiceConfiguration",
+		    "route53:ListHostedZonesByVPC",
+		    "route53:CreateVPCAssociationAuthorization",
+		    "route53:AssociateVPCWithHostedZone",
+		    "route53:DeleteVPCAssociationAuthorization",
+		    "route53:DisassociateVPCFromHostedZone"
```
Comment (oriAdler) [inline resources/sts/4.11/sts_installer_privatelink_permission_policy.json:12]: Should we add all these new permissions to the sts_installer_permission_policy.json as well?
To have the installer policy equivalent to the three new policies?

**Comment by oriAdler** [comment]
> @oriAdler lets remove the permissions that aren't in the current `sts_installer_permission_policy.json` file so we can get this merged and we'll investigate the missing permissions and add them if needed.

I removed the private link policy since all permissions in this file were new.
Let me create a private link cluster to verify a successful installation.

**Comment by oriAdler** [comment]
@jharrington22 I was able to create a private link cluster with this policy `resources/sts/4.11/sts_installer_core_permission_policy.json` attached to the installer role.

File: resources/sts/4.11/sts_installer_vpc_permission_policy.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		    "ec2:CreateDhcpOptions",
+		    "ec2:CreateInternetGateway",
+		    "ec2:CreateNatGateway",
+		    "ec2:CreateRouteTable",
+		    "ec2:CreateSubnet",
+		    "ec2:CreateVpc",
+		    "ec2:DeleteDhcpOptions",
+		    "ec2:DeleteInternetGateway",
+		    "ec2:DeleteNatGateway",
+		    "ec2:DeleteNetworkInterface",
```
Comment (jharrington22) [inline resources/sts/4.11/sts_installer_vpc_permission_policy.json:19]: We can remove this its a duplicate from the core policy

File: resources/sts/4.11/sts_installer_vpc_permission_policy.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		    "ec2:DeleteRoute",
```
Comment (jharrington22) [inline resources/sts/4.11/sts_installer_vpc_permission_policy.json:20]: This is also a duplicate

File: resources/sts/4.11/sts_installer_core_permission_policy.json [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+		    "ec2:DescribeSecurityGroups",
+		    "ec2:DescribeSubnets",
+		    "ec2:DescribeTags",
+		    "ec2:DescribeVolumes",
+		    "ec2:DescribeVpcAttribute",
+		    "ec2:DescribeVpcClassicLink",
+		    "ec2:DescribeVpcClassicLinkDnsSupport",
+		    "ec2:DescribeVpcEndpoints",
+		    "ec2:DescribeVpcs",
+		    "ec2:DisassociateRouteTable",
```
Comment (jharrington22) [inline resources/sts/4.11/sts_installer_core_permission_policy.json:53]: We can remove this, it should only be required for VPCs

**Review by jharrington22 — CHANGES_REQUESTED** [review]

**Comment by jharrington22** [comment]
@oriAdler `GetBucketReplication` is also missing from this but it does exist in the [sts_installer_permission_policy.json](https://github.com/openshift/managed-cluster-config/blob/master/resources/sts/4.11/sts_installer_permission_policy.json#L154)

File: resources/sts/4.11/sts_installer_vpc_permission_policy.json (hunk 2 above)
Comment (oriAdler) [reply resources/sts/4.11/sts_installer_vpc_permission_policy.json]: Removed.

File: resources/sts/4.11/sts_installer_vpc_permission_policy.json (hunk 3 above)
Comment (oriAdler) [reply resources/sts/4.11/sts_installer_vpc_permission_policy.json]: Removed.

File: resources/sts/4.11/sts_installer_core_permission_policy.json (hunk 4 above)
Comment (oriAdler) [reply resources/sts/4.11/sts_installer_core_permission_policy.json]: Removed.

**Comment by oriAdler** [comment]
> @oriAdler `GetBucketReplication` is also missing from this but it does exist in the [sts_installer_permission_policy.json](https://github.com/openshift/managed-cluster-config/blob/master/resources/sts/4.11/sts_installer_permission_policy.json#L154)

Added.

**Comment by oriAdler** [comment]
@fahlmant can you PTAL?

**Comment by jharrington22** [comment]
@oriAdler this isn't merging because we're blocking merges to master with #1395

**Review by jharrington22 — APPROVED** [review]

**Comment by jharrington22** [comment]
Merged during change freeze, see https://issues.redhat.com/browse/SDA-6771 for more details

**Pull request merged**